CONTRIBUTING.asciidoc

```diff
@@ -0,0 +1,356 @@
+= Contributing to ipa
+
+=== Installation
+
+[source]
+----
+git clone https://github.com/SUSE/ipa.git
+cd ipa
+
+# Activate virtual Environment then install
+# ipa and dev dependences in editable mode
+pip install -e .[dev]
+----
+
+ipa is now installed in the active virtual environment in development
+mode.
+
+=== Dev Requirements
+
+* bumpversion
+* pip>=7.0.0
+
+=== Testing Requirements
+
+* coverage
+* flake8
+* mock
+* pytest-cov
+* vcrpy
+
+=== Versions & Releases
+
+*ipa* adheres to Semantic versioning; see http://semver.org/ for details.
+
+link:https://pypi.python.org/pypi/bumpversion/[bumpversion] is used
+for release version management, and is configured in `setup.cfg`:
+
+----
+bumpversion major|minor|patch
+git push && git push --tags
+----
+
+=== Running the tests
+
+All tests should pass and test coverage should remain above 90%.
+
+The tests and coverage can be run directly via Pytest.
+
+----
+pytest --cov=ipa
+----
+
+They can also be run with tox integration which runs flake8 as well:
+
+=== Testing with tox
+
+==== Requirements
+
+* tox
+* tox-pyenv (testing multiple Python envs)
+
+==== Tox installation
+
+pip install .[tox]
+
+==== Setup pyenv
+
+See the link:https://github.com/pyenv/pyenv#installation[pyenv doc] for more
+info.
+
+==== Run tox
+
+----
+tox (test all versions)
+
+# or
+tox -e py{version}
+tox -e py36
+----
+
+=== Code Style
+
+Source should pass flake8 and pep8 standards.
+
+----
+flake8 ipa
+----
+
+== ipa Source Overview
+
```

Comment:
I'll pull the the source overview and test information from this request and add a simple overview in readme.

I think moving the in depth overview and examples to a structured docs folder would be more helpful. Thus I'll make a separate request with the docs.